src/common/arrayparser.rs

```diff
@@ -1,25 +1,46 @@
 use anyhow::anyhow;
 use nom::bytes::complete::tag;
 use nom::combinator::all_consuming;
-use nom::combinator::{map, map_res, opt};
+use nom::combinator::{map_res, opt};
 use nom::sequence::{preceded, tuple};
 
 use crate::common::arraydef::{ArrayDef, TaskIdRange};
 use crate::common::parser::{p_uint, NomResult};
+use crate::Set;
+use nom::multi::fold_many0;
 
 fn p_task_id_range(input: &str) -> NomResult<TaskIdRange> {
     map_res(
-        tuple((p_uint, opt(preceded(tag("-"), p_uint)))),
+        tuple((
+            preceded(tag(","), p_uint),
+            opt(preceded(tag("-"), p_uint)),
+            opt(preceded(tag(":"), p_uint)),
+        )),
         |r| match r {
-            (v, None) => Ok(TaskIdRange::new(v, 1)),
-            (v, Some(w)) if w >= v => Ok(TaskIdRange::new(v, w - v + 1)),
+            (v, None, None) => Ok(TaskIdRange::new(v, 1, 1)),
+            (v, Some(w), None) if w >= v => Ok(TaskIdRange::new(v, w - v + 1, 1)),
+            (v, Some(w), Some(x)) if w >= v && x <= w - v => Ok(TaskIdRange::new(v, w - v + 1, x)),
             _ => Err("Invalid range"),
         },
     )(input)
 }
 
 fn p_array_def(input: &str) -> NomResult<ArrayDef> {
-    map(p_task_id_range, ArrayDef::new)(input)
+    map_res(p_task_id_ranges, |r| match r {
+        res if !is_overlapping(res.clone()) => Ok(ArrayDef::new(res)),
+        _ => Err("Ranges overlap"),
```

Comment:
@spirali Why should overlapping ranges cause an error? Maybe a warning could be printed, but it seems to me that an error is too strict.

Reply:
I prefer the error. I do not see a usecase.

Reply:
It seems to me that overlap is totally harmless here.

It's an artificial example, but if I want to e.g. resubmit every even task and also tasks from some range, I could just do
`0-1000:2, 500-600`. With this check I would need to split the intervals manually.